contracts/src/v0.8/vrf/test/BatchVRFCoordinatorV2Plus.t.sol

```diff
@@ -0,0 +1,178 @@
+
+    // Proof generated via the generate-proof-v2-plus script command. Example usage:
+    // _printGenerateProofV2PlusCommand(address(s_consumer1), 1, requestBlock, false);
+    /*
+       go run . generate-proof-v2-plus \
+        -key-hash 0x9f2353bde94264dbc3d554a94cceba2d7d2b4fdce4304d3e09a1fea9fbeb1528 \
+        -pre-seed 76568185840201037774581758921393822690942290841865097674309745036496166431060 \
+        -block-hash 0x14 \
+        -block-num 20 \
+        -sender 0x2f1c0761d6e4b1e5f01968d6c746f695e5f3e25d \
+        -native-payment false
+    */
+    proofs[1] = VRFTypes.Proof({
+      pk: [
+        72488970228380509287422715226575535698893157273063074627791787432852706183111,
+        62070622898698443831883535403436258712770888294397026493185421712108624767191
+      ],
+      gamma: [
+        21323932463597506192387578758854201988004673105893105492473194972397109828006,
+        96834737826889397196571646974355352644437196500310392203712129010026003355112
+      ],
+      c: 8775807990949224376582975115621037245862755412370175152581490650310350359728,
+      s: 6805708577951013810918872616271445638109899206333819877111740872779453350091,
+      seed: 76568185840201037774581758921393822690942290841865097674309745036496166431060,
+      uWitness: 0xE82fF24Fecfbe73d682f38308bE3E039Dfabdf5c,
+      cGammaWitness: [
+        92810770919624535241476539842820168209710445519252592382122118536598338376923,
+        17271305664006119131434661141858450289379246199095231636439133258170648418554
+      ],
+      sHashWitness: [
+        29540023305939374439696120003978246982707698669656874393367212257432197207536,
+        93902323936532381028323379401739289810874348405259732508442252936582467730050
+      ],
+      zInv: 88845170436601946907659333156418518556235340365885668267853966404617557948692
+    });
+    rcs[1] = VRFTypes.RequestCommitmentV2Plus({
+      blockNum: requestBlock,
+      subId: s_subId,
+      callbackGasLimit: CALLBACK_GAS_LIMIT,
+      numWords: 1,
+      sender: address(s_consumer1),
+      extraArgs: VRFV2PlusClient._argsToBytes(VRFV2PlusClient.ExtraArgsV1({nativePayment: false}))
+    });
+
+    VRFCoordinatorV2_5.Output memory output1 = s_coordinator.getRandomnessFromProofExternal(
+      abi.decode(abi.encode(proofs[1]), (VRF.Proof)),
+      rcs[1]
+    );
+
+    // The payments are NOT pre-calculated and simply copied from the actual event.
+    // We can assert and ignore the payment field but the code will be considerably longer.
+    vm.expectEmit(true, true, false, true, address(s_coordinator));
+    emit RandomWordsFulfilled(output.requestId, output.randomness, s_subId, 500000000000142215, true, true, false);
+    vm.expectEmit(true, true, false, true, address(s_coordinator));
+    emit RandomWordsFulfilled(output1.requestId, output1.randomness, s_subId, 800000000000304103, false, true, false);
+
+    // Fulfill the requests.
+    s_batchCoordinator.fulfillRandomWords(proofs, rcs);
```

Comment:
To be honest, the main motivation for the test is to have some code from the test to trigger the `fulfillRandomWords` so that I can get its gas usage. 😅

So right now, the assertion is quite minimal and it just asserts that the 2 requests are fulfilled.

On hindsight, I don't think we should be doing extensive assertions here as well, as subscription balances check should be done by the coordinator itself. The batch coordinator's job is just to proxy over to the coordinator and fulfill multiple requests. We could do those checks here as well but it will really be a redundant check already covered by the coordinator. If the logic were to change, then we'll need to fix multiple places as well.